File: java_sdk/src/main/java/business_api_client/CampaignCreateBody.java
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */

package business_api_client;

import java.util.Objects;
import java.util.Arrays;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.ArrayList;
import java.util.List;
/**
 * CampaignCreateBody
 */

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaClientCodegen", date = "2026-01-08T15:25:30.628143-08:00[America/Los_Angeles]")
public class CampaignCreateBody {
  @JsonProperty("advertiser_id")
  private String advertiserId = null;

  @JsonProperty("app_id")
  private String appId = null;

  @JsonProperty("app_promotion_type")
  private String appPromotionType = null;

  @JsonProperty("bid_type")
  private String bidType = null;

  @JsonProperty("budget")
  private Float budget = null;

  @JsonProperty("budget_mode")
  private String budgetMode = null;

  @JsonProperty("budget_optimize_on")
  private Boolean budgetOptimizeOn = false;

  @JsonProperty("campaign_app_profile_page_state")
  private String campaignAppProfilePageState = null;

  @JsonProperty("campaign_name")
  private String campaignName = null;

  @JsonProperty("campaign_product_source")
  private String campaignProductSource = null;

  @JsonProperty("campaign_type")
  private String campaignType = null;

  @JsonProperty("catalog_enabled")
  private Boolean catalogEnabled = null;

  @JsonProperty("deep_bid_type")
  private String deepBidType = null;

  @JsonProperty("disable_skan_campaign")
  private Boolean disableSkanCampaign = null;

  @JsonProperty("internal_channel")
  private String internalChannel = null;

  @JsonProperty("is_advanced_dedicated_campaign")
  private Boolean isAdvancedDedicatedCampaign = null;

  @JsonProperty("is_search_campaign")
  private Boolean isSearchCampaign = null;

  @JsonProperty("objective_type")
  private String objectiveType = null;

  @JsonProperty("operation_status")
  private String operationStatus = "ENABLE";

  @JsonProperty("optimization_goal")
  private String optimizationGoal = null;

  @JsonProperty("plugin_partner")
  private String pluginPartner = null;

  @JsonProperty("po_number")
  private String poNumber = null;

  @JsonProperty("postback_window_mode")
  private String postbackWindowMode = null;

  @JsonProperty("request_id")
  private String requestId = null;

  @JsonProperty("rf_campaign_type")
  private String rfCampaignType = null;

  @JsonProperty("roas_bid")
  private Float roasBid = null;

  @JsonProperty("rta_bid_enabled")
  private Boolean rtaBidEnabled = false;

  @JsonProperty("rta_id")
  private String rtaId = null;

  @JsonProperty("rta_product_selection_enabled")
  private Boolean rtaProductSelectionEnabled = null;

  @JsonProperty("sales_destination")
  private String salesDestination = null;

  @JsonProperty("special_industries")
  private List<String> specialIndustries = null;

  @JsonProperty("virtual_objective_type")
  private String virtualObjectiveType = null;

  public CampaignCreateBody advertiserId(String advertiserId) {
    this.advertiserId = advertiserId;
    return this;
  }

   /**
   * Get advertiserId
   * @return advertiserId
  **/
  @Schema(required = true, description = "")
  public String getAdvertiserId() {
    return advertiserId;
  }

  public void setAdvertiserId(String advertiserId) {
    this.advertiserId = advertiserId;
  }

  public CampaignCreateBody appId(String appId) {
    this.appId = appId;
    return this;
  }

   /**
   * Get appId
   * @return appId
  **/
  @Schema(description = "")
  public String getAppId() {
    return appId;
  }

  public void setAppId(String appId) {
    this.appId = appId;
  }

  public CampaignCreateBody appPromotionType(String appPromotionType) {
    this.appPromotionType = appPromotionType;
    return this;
  }

   /**
   * Get appPromotionType
   * @return appPromotionType
  **/
  @Schema(description = "")
  public String getAppPromotionType() {
    return appPromotionType;
  }

  public void setAppPromotionType(String appPromotionType) {
    this.appPromotionType = appPromotionType;
  }

  public CampaignCreateBody bidType(String bidType) {
    this.bidType = bidType;
    return this;
  }

   /**
   * Get bidType
   * @return bidType
  **/
  @Schema(description = "")
  public String getBidType() {
    return bidType;
  }

  public void setBidType(String bidType) {
    this.bidType = bidType;
  }

  public CampaignCreateBody budget(Float budget) {
    this.budget = budget;
    return this;
  }

   /**
   * Get budget
   * @return budget
  **/
  @Schema(description = "")
  public Float getBudget() {
    return budget;
  }

  public void setBudget(Float budget) {
    this.budget = budget;
  }

  public CampaignCreateBody budgetMode(String budgetMode) {
    this.budgetMode = budgetMode;
    return this;
  }

   /**
   * Get budgetMode
   * @return budgetMode
  **/
  @Schema(description = "")
  public String getBudgetMode() {
    return budgetMode;
  }

  public void setBudgetMode(String budgetMode) {
    this.budgetMode = budgetMode;
  }

  public CampaignCreateBody budgetOptimizeOn(Boolean budgetOptimizeOn) {
    this.budgetOptimizeOn = budgetOptimizeOn;
    return this;
  }

   /**
   * Get budgetOptimizeOn
   * @return budgetOptimizeOn
  **/
  @Schema(description = "")
  public Boolean isBudgetOptimizeOn() {
    return budgetOptimizeOn;
  }

  public void setBudgetOptimizeOn(Boolean budgetOptimizeOn) {
    this.budgetOptimizeOn = budgetOptimizeOn;
  }

  public CampaignCreateBody campaignAppProfilePageState(String campaignAppProfilePageState) {
    this.campaignAppProfilePageState = campaignAppProfilePageState;
    return this;
  }

   /**
   * Get campaignAppProfilePageState
   * @return campaignAppProfilePageState
  **/
  @Schema(description = "")
  public String getCampaignAppProfilePageState() {
    return campaignAppProfilePageState;
  }

  public void setCampaignAppProfilePageState(String campaignAppProfilePageState) {
    this.campaignAppProfilePageState = campaignAppProfilePageState;
  }

  public CampaignCreateBody campaignName(String campaignName) {
    this.campaignName = campaignName;
    return this;
  }

   /**
   * Get campaignName
   * @return campaignName
  **/
  @Schema(required = true, description = "")
  public String getCampaignName() {
    return campaignName;
  }

  public void setCampaignName(String campaignName) {
    this.campaignName = campaignName;
  }

  public CampaignCreateBody campaignProductSource(String campaignProductSource) {
    this.campaignProductSource = campaignProductSource;
    return this;
  }

   /**
   * Get campaignProductSource
   * @return campaignProductSource
  **/
  @Schema(description = "")
  public String getCampaignProductSource() {
    return campaignProductSource;
  }

  public void setCampaignProductSource(String campaignProductSource) {
    this.campaignProductSource = campaignProductSource;
  }

  public CampaignCreateBody campaignType(String campaignType) {
    this.campaignType = campaignType;
    return this;
  }

   /**
   * Get campaignType
   * @return campaignType
  **/
  @Schema(description = "")
  public String getCampaignType() {
    return campaignType;
  }

  public void setCampaignType(String campaignType) {
    this.campaignType = campaignType;
  }

  public CampaignCreateBody catalogEnabled(Boolean catalogEnabled) {
    this.catalogEnabled = catalogEnabled;
    return this;
  }

   /**
   * Get catalogEnabled
   * @return catalogEnabled
  **/
  @Schema(description = "")
  public Boolean isCatalogEnabled() {
    return catalogEnabled;
  }

  public void setCatalogEnabled(Boolean catalogEnabled) {
    this.catalogEnabled = catalogEnabled;
  }

  public CampaignCreateBody deepBidType(String deepBidType) {
    this.deepBidType = deepBidType;
    return this;
  }

   /**
   * Get deepBidType
   * @return deepBidType
  **/
  @Schema(description = "")
  public String getDeepBidType() {
    return deepBidType;
  }

  public void setDeepBidType(String deepBidType) {
    this.deepBidType = deepBidType;
  }

  public CampaignCreateBody disableSkanCampaign(Boolean disableSkanCampaign) {
    this.disableSkanCampaign = disableSkanCampaign;
    return this;
  }

   /**
   * Get disableSkanCampaign
   * @return disableSkanCampaign
  **/
  @Schema(description = "")
  public Boolean isDisableSkanCampaign() {
    return disableSkanCampaign;
  }

  public void setDisableSkanCampaign(Boolean disableSkanCampaign) {
    this.disableSkanCampaign = disableSkanCampaign;
  }

  public CampaignCreateBody internalChannel(String internalChannel) {
    this.internalChannel = internalChannel;
    return this;
  }

   /**
   * Get internalChannel
   * @return internalChannel
  **/
  @Schema(description = "")
  public String getInternalChannel() {
    return internalChannel;
  }

  public void setInternalChannel(String internalChannel) {
    this.internalChannel = internalChannel;
  }

  public CampaignCreateBody isAdvancedDedicatedCampaign(Boolean isAdvancedDedicatedCampaign) {
    this.isAdvancedDedicatedCampaign = isAdvancedDedicatedCampaign;
    return this;
  }

   /**
   * Get isAdvancedDedicatedCampaign
   * @return isAdvancedDedicatedCampaign
  **/
  @Schema(description = "")
  public Boolean isIsAdvancedDedicatedCampaign() {
    return isAdvancedDedicatedCampaign;
  }

  public void setIsAdvancedDedicatedCampaign(Boolean isAdvancedDedicatedCampaign) {
    this.isAdvancedDedicatedCampaign = isAdvancedDedicatedCampaign;
  }

  public CampaignCreateBody isSearchCampaign(Boolean isSearchCampaign) {
    this.isSearchCampaign = isSearchCampaign;
    return this;
  }

   /**
   * Get isSearchCampaign
   * @return isSearchCampaign
  **/
  @Schema(description = "")
  public Boolean isIsSearchCampaign() {
    return isSearchCampaign;
  }

  public void setIsSearchCampaign(Boolean isSearchCampaign) {
    this.isSearchCampaign = isSearchCampaign;
  }

  public CampaignCreateBody objectiveType(String objectiveType) {
    this.objectiveType = objectiveType;
    return this;
  }

   /**
   * Get objectiveType
   * @return objectiveType
  **/
  @Schema(required = true, description = "")
  public String getObjectiveType() {
    return objectiveType;
  }

  public void setObjectiveType(String objectiveType) {
    this.objectiveType = objectiveType;
  }

  public CampaignCreateBody operationStatus(String operationStatus) {
    this.operationStatus = operationStatus;
    return this;
  }

   /**
   * Get operationStatus
   * @return operationStatus
  **/
  @Schema(description = "")
  public String getOperationStatus() {
    return operationStatus;
  }

  public void setOperationStatus(String operationStatus) {
    this.operationStatus = operationStatus;
  }

  public CampaignCreateBody optimizationGoal(String optimizationGoal) {
    this.optimizationGoal = optimizationGoal;
    return this;
  }

   /**
   * Get optimizationGoal
   * @return optimizationGoal
  **/
  @Schema(description = "")
  public String getOptimizationGoal() {
    return optimizationGoal;
  }

  public void setOptimizationGoal(String optimizationGoal) {
    this.optimizationGoal = optimizationGoal;
  }

  public CampaignCreateBody pluginPartner(String pluginPartner) {
    this.pluginPartner = pluginPartner;
    return this;
  }

   /**
   * Get pluginPartner
   * @return pluginPartner
  **/
  @Schema(description = "")
  public String getPluginPartner() {
    return pluginPartner;
  }

  public void setPluginPartner(String pluginPartner) {
    this.pluginPartner = pluginPartner;
  }

  public CampaignCreateBody poNumber(String poNumber) {
    this.poNumber = poNumber;
    return this;
  }

   /**
   * Get poNumber
   * @return poNumber
  **/
  @Schema(description = "")
  public String getPoNumber() {
    return poNumber;
  }

  public void setPoNumber(String poNumber) {
    this.poNumber = poNumber;
  }

  public CampaignCreateBody postbackWindowMode(String postbackWindowMode) {
    this.postbackWindowMode = postbackWindowMode;
    return this;
  }

   /**
   * Get postbackWindowMode
   * @return postbackWindowMode
  **/
  @Schema(description = "")
  public String getPostbackWindowMode() {
    return postbackWindowMode;
  }

  public void setPostbackWindowMode(String postbackWindowMode) {
    this.postbackWindowMode = postbackWindowMode;
  }

  public CampaignCreateBody requestId(String requestId) {
    this.requestId = requestId;
    return this;
  }

   /**
   * Get requestId
   * @return requestId
  **/
  @Schema(description = "")
  public String getRequestId() {
    return requestId;
  }

  public void setRequestId(String requestId) {
    this.requestId = requestId;
  }

  public CampaignCreateBody rfCampaignType(String rfCampaignType) {
    this.rfCampaignType = rfCampaignType;
    return this;
  }

   /**
   * Get rfCampaignType
   * @return rfCampaignType
  **/
  @Schema(description = "")
  public String getRfCampaignType() {
    return rfCampaignType;
  }

  public void setRfCampaignType(String rfCampaignType) {
    this.rfCampaignType = rfCampaignType;
  }

  public CampaignCreateBody roasBid(Float roasBid) {
    this.roasBid = roasBid;
    return this;
  }

   /**
   * Get roasBid
   * @return roasBid
  **/
  @Schema(description = "")
  public Float getRoasBid() {
    return roasBid;
  }

  public void setRoasBid(Float roasBid) {
    this.roasBid = roasBid;
  }

  public CampaignCreateBody rtaBidEnabled(Boolean rtaBidEnabled) {
    this.rtaBidEnabled = rtaBidEnabled;
    return this;
  }

   /**
   * Get rtaBidEnabled
   * @return rtaBidEnabled
  **/
  @Schema(description = "")
  public Boolean isRtaBidEnabled() {
    return rtaBidEnabled;
  }

  public void setRtaBidEnabled(Boolean rtaBidEnabled) {
    this.rtaBidEnabled = rtaBidEnabled;
  }

  public CampaignCreateBody rtaId(String rtaId) {
    this.rtaId = rtaId;
    return this;
  }

   /**
   * Get rtaId
   * @return rtaId
  **/
  @Schema(description = "")
  public String getRtaId() {
    return rtaId;
  }

  public void setRtaId(String rtaId) {
    this.rtaId = rtaId;
  }

  public CampaignCreateBody rtaProductSelectionEnabled(Boolean rtaProductSelectionEnabled) {
    this.rtaProductSelectionEnabled = rtaProductSelectionEnabled;
    return this;
  }

   /**
   * Get rtaProductSelectionEnabled
   * @return rtaProductSelectionEnabled
  **/
  @Schema(description = "")
  public Boolean isRtaProductSelectionEnabled() {
    return rtaProductSelectionEnabled;
  }

  public void setRtaProductSelectionEnabled(Boolean rtaProductSelectionEnabled) {
    this.rtaProductSelectionEnabled = rtaProductSelectionEnabled;
  }

  public CampaignCreateBody salesDestination(String salesDestination) {
    this.salesDestination = salesDestination;
    return this;
  }

   /**
   * Get salesDestination
   * @return salesDestination
  **/
  @Schema(description = "")
  public String getSalesDestination() {
    return salesDestination;
  }

  public void setSalesDestination(String salesDestination) {
    this.salesDestination = salesDestination;
  }

  public CampaignCreateBody specialIndustries(List<String> specialIndustries) {
    this.specialIndustries = specialIndustries;
    return this;
  }

  public CampaignCreateBody addSpecialIndustriesItem(String specialIndustriesItem) {
    if (this.specialIndustries == null) {
      this.specialIndustries = new ArrayList<String>();
    }
    this.specialIndustries.add(specialIndustriesItem);
    return this;
  }

   /**
   * Get specialIndustries
   * @return specialIndustries
  **/
  @Schema(description = "")
  public List<String> getSpecialIndustries() {
    return specialIndustries;
  }

  public void setSpecialIndustries(List<String> specialIndustries) {
    this.specialIndustries = specialIndustries;
  }

  public CampaignCreateBody virtualObjectiveType(String virtualObjectiveType) {
    this.virtualObjectiveType = virtualObjectiveType;
    return this;
  }

   /**
   * Get virtualObjectiveType
   * @return virtualObjectiveType
  **/
  @Schema(description = "")
  public String getVirtualObjectiveType() {
    return virtualObjectiveType;
  }

  public void setVirtualObjectiveType(String virtualObjectiveType) {
    this.virtualObjectiveType = virtualObjectiveType;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    CampaignCreateBody campaignCreateBody = (CampaignCreateBody) o;
    return Objects.equals(this.advertiserId, campaignCreateBody.advertiserId) &&
        Objects.equals(this.appId, campaignCreateBody.appId) &&
        Objects.equals(this.appPromotionType, campaignCreateBody.appPromotionType) &&
        Objects.equals(this.bidType, campaignCreateBody.bidType) &&
        Objects.equals(this.budget, campaignCreateBody.budget) &&
        Objects.equals(this.budgetMode, campaignCreateBody.budgetMode) &&
        Objects.equals(this.budgetOptimizeOn, campaignCreateBody.budgetOptimizeOn) &&
        Objects.equals(this.campaignAppProfilePageState, campaignCreateBody.campaignAppProfilePageState) &&
        Objects.equals(this.campaignName, campaignCreateBody.campaignName) &&
        Objects.equals(this.campaignProductSource, campaignCreateBody.campaignProductSource) &&
        Objects.equals(this.campaignType, campaignCreateBody.campaignType) &&
        Objects.equals(this.catalogEnabled, campaignCreateBody.catalogEnabled) &&
        Objects.equals(this.deepBidType, campaignCreateBody.deepBidType) &&
        Objects.equals(this.disableSkanCampaign, campaignCreateBody.disableSkanCampaign) &&
        Objects.equals(this.internalChannel, campaignCreateBody.internalChannel) &&
        Objects.equals(this.isAdvancedDedicatedCampaign, campaignCreateBody.isAdvancedDedicatedCampaign) &&
        Objects.equals(this.isSearchCampaign, campaignCreateBody.isSearchCampaign) &&
        Objects.equals(this.objectiveType, campaignCreateBody.objectiveType) &&
        Objects.equals(this.operationStatus, campaignCreateBody.operationStatus) &&
        Objects.equals(this.optimizationGoal, campaignCreateBody.optimizationGoal) &&
        Objects.equals(this.pluginPartner, campaignCreateBody.pluginPartner) &&
        Objects.equals(this.poNumber, campaignCreateBody.poNumber) &&
        Objects.equals(this.postbackWindowMode, campaignCreateBody.postbackWindowMode) &&
        Objects.equals(this.requestId, campaignCreateBody.requestId) &&
        Objects.equals(this.rfCampaignType, campaignCreateBody.rfCampaignType) &&
        Objects.equals(this.roasBid, campaignCreateBody.roasBid) &&
        Objects.equals(this.rtaBidEnabled, campaignCreateBody.rtaBidEnabled) &&
        Objects.equals(this.rtaId, campaignCreateBody.rtaId) &&
        Objects.equals(this.rtaProductSelectionEnabled, campaignCreateBody.rtaProductSelectionEnabled) &&
        Objects.equals(this.salesDestination, campaignCreateBody.salesDestination) &&
        Objects.equals(this.specialIndustries, campaignCreateBody.specialIndustries) &&
        Objects.equals(this.virtualObjectiveType, campaignCreateBody.virtualObjectiveType);
  }

  @Override
  public int hashCode() {
    return Objects.hash(advertiserId, appId, appPromotionType, bidType, budget, budgetMode, budgetOptimizeOn, campaignAppProfilePageState, campaignName, campaignProductSource, campaignType, catalogEnabled, deepBidType, disableSkanCampaign, internalChannel, isAdvancedDedicatedCampaign, isSearchCampaign, objectiveType, operationStatus, optimizationGoal, pluginPartner, poNumber, postbackWindowMode, requestId, rfCampaignType, roasBid, rtaBidEnabled, rtaId, rtaProductSelectionEnabled, salesDestination, specialIndustries, virtualObjectiveType);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class CampaignCreateBody {\n");
    
    sb.append("    advertiserId: ").append(toIndentedString(advertiserId)).append("\n");
    sb.append("    appId: ").append(toIndentedString(appId)).append("\n");
    sb.append("    appPromotionType: ").append(toIndentedString(appPromotionType)).append("\n");
    sb.append("    bidType: ").append(toIndentedString(bidType)).append("\n");
    sb.append("    budget: ").append(toIndentedString(budget)).append("\n");
    sb.append("    budgetMode: ").append(toIndentedString(budgetMode)).append("\n");
    sb.append("    budgetOptimizeOn: ").append(toIndentedString(budgetOptimizeOn)).append("\n");
    sb.append("    campaignAppProfilePageState: ").append(toIndentedString(campaignAppProfilePageState)).append("\n");
    sb.append("    campaignName: ").append(toIndentedString(campaignName)).append("\n");
    sb.append("    campaignProductSource: ").append(toIndentedString(campaignProductSource)).append("\n");
    sb.append("    campaignType: ").append(toIndentedString(campaignType)).append("\n");
    sb.append("    catalogEnabled: ").append(toIndentedString(catalogEnabled)).append("\n");
    sb.append("    deepBidType: ").append(toIndentedString(deepBidType)).append("\n");
    sb.append("    disableSkanCampaign: ").append(toIndentedString(disableSkanCampaign)).append("\n");
    sb.append("    internalChannel: ").append(toIndentedString(internalChannel)).append("\n");
    sb.append("    isAdvancedDedicatedCampaign: ").append(toIndentedString(isAdvancedDedicatedCampaign)).append("\n");
    sb.append("    isSearchCampaign: ").append(toIndentedString(isSearchCampaign)).append("\n");
    sb.append("    objectiveType: ").append(toIndentedString(objectiveType)).append("\n");
    sb.append("    operationStatus: ").append(toIndentedString(operationStatus)).append("\n");
    sb.append("    optimizationGoal: ").append(toIndentedString(optimizationGoal)).append("\n");
    sb.append("    pluginPartner: ").append(toIndentedString(pluginPartner)).append("\n");
    sb.append("    poNumber: ").append(toIndentedString(poNumber)).append("\n");
    sb.append("    postbackWindowMode: ").append(toIndentedString(postbackWindowMode)).append("\n");
    sb.append("    requestId: ").append(toIndentedString(requestId)).append("\n");
    sb.append("    rfCampaignType: ").append(toIndentedString(rfCampaignType)).append("\n");
    sb.append("    roasBid: ").append(toIndentedString(roasBid)).append("\n");
    sb.append("    rtaBidEnabled: ").append(toIndentedString(rtaBidEnabled)).append("\n");
    sb.append("    rtaId: ").append(toIndentedString(rtaId)).append("\n");
    sb.append("    rtaProductSelectionEnabled: ").append(toIndentedString(rtaProductSelectionEnabled)).append("\n");
    sb.append("    salesDestination: ").append(toIndentedString(salesDestination)).append("\n");
    sb.append("    specialIndustries: ").append(toIndentedString(specialIndustries)).append("\n");
    sb.append("    virtualObjectiveType: ").append(toIndentedString(virtualObjectiveType)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

File: js_sdk/src/model/CampaignCreateBody.js
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
import {ApiClient} from '../ApiClient.js';

/**
 * The CampaignCreateBody model module.
 * @module model/CampaignCreateBody
 * @version 1.2.1
 */
export class CampaignCreateBody {
  /**
   * Constructs a new <code>CampaignCreateBody</code>.
   * @alias module:model/CampaignCreateBody
   * @class
   * @param advertiser_id {String} 
   * @param campaign_name {String} 
   * @param objective_type {String} 
   */
  constructor(advertiser_id, campaign_name, objective_type) {
    this.advertiser_id = advertiser_id;
    this.campaign_name = campaign_name;
    this.objective_type = objective_type;
  }

  /**
   * Constructs a <code>CampaignCreateBody</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/CampaignCreateBody} obj Optional instance to populate.
   * @return {module:model/CampaignCreateBody} The populated <code>CampaignCreateBody</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new CampaignCreateBody();
      if (data.hasOwnProperty('advertiser_id'))
        obj.advertiser_id = ApiClient.convertToType(data['advertiser_id'], 'String');
      if (data.hasOwnProperty('app_id'))
        obj.app_id = ApiClient.convertToType(data['app_id'], 'String');
      if (data.hasOwnProperty('app_promotion_type'))
        obj.app_promotion_type = ApiClient.convertToType(data['app_promotion_type'], 'String');
      if (data.hasOwnProperty('bid_type'))
        obj.bid_type = ApiClient.convertToType(data['bid_type'], 'String');
      if (data.hasOwnProperty('budget'))
        obj.budget = ApiClient.convertToType(data['budget'], 'Number');
      if (data.hasOwnProperty('budget_mode'))
        obj.budget_mode = ApiClient.convertToType(data['budget_mode'], 'String');
      if (data.hasOwnProperty('budget_optimize_on'))
        obj.budget_optimize_on = ApiClient.convertToType(data['budget_optimize_on'], 'Boolean');
      if (data.hasOwnProperty('campaign_app_profile_page_state'))
        obj.campaign_app_profile_page_state = ApiClient.convertToType(data['campaign_app_profile_page_state'], 'String');
      if (data.hasOwnProperty('campaign_name'))
        obj.campaign_name = ApiClient.convertToType(data['campaign_name'], 'String');
      if (data.hasOwnProperty('campaign_product_source'))
        obj.campaign_product_source = ApiClient.convertToType(data['campaign_product_source'], 'String');
      if (data.hasOwnProperty('campaign_type'))
        obj.campaign_type = ApiClient.convertToType(data['campaign_type'], 'String');
      if (data.hasOwnProperty('catalog_enabled'))
        obj.catalog_enabled = ApiClient.convertToType(data['catalog_enabled'], 'Boolean');
      if (data.hasOwnProperty('deep_bid_type'))
        obj.deep_bid_type = ApiClient.convertToType(data['deep_bid_type'], 'String');
      if (data.hasOwnProperty('disable_skan_campaign'))
        obj.disable_skan_campaign = ApiClient.convertToType(data['disable_skan_campaign'], 'Boolean');
      if (data.hasOwnProperty('internal_channel'))
        obj.internal_channel = ApiClient.convertToType(data['internal_channel'], 'String');
      if (data.hasOwnProperty('is_advanced_dedicated_campaign'))
        obj.is_advanced_dedicated_campaign = ApiClient.convertToType(data['is_advanced_dedicated_campaign'], 'Boolean');
      if (data.hasOwnProperty('is_search_campaign'))
        obj.is_search_campaign = ApiClient.convertToType(data['is_search_campaign'], 'Boolean');
      if (data.hasOwnProperty('objective_type'))
        obj.objective_type = ApiClient.convertToType(data['objective_type'], 'String');
      if (data.hasOwnProperty('operation_status'))
        obj.operation_status = ApiClient.convertToType(data['operation_status'], 'String');
      if (data.hasOwnProperty('optimization_goal'))
        obj.optimization_goal = ApiClient.convertToType(data['optimization_goal'], 'String');
      if (data.hasOwnProperty('plugin_partner'))
        obj.plugin_partner = ApiClient.convertToType(data['plugin_partner'], 'String');
      if (data.hasOwnProperty('po_number'))
        obj.po_number = ApiClient.convertToType(data['po_number'], 'String');
      if (data.hasOwnProperty('postback_window_mode'))
        obj.postback_window_mode = ApiClient.convertToType(data['postback_window_mode'], 'String');
      if (data.hasOwnProperty('request_id'))
        obj.request_id = ApiClient.convertToType(data['request_id'], 'String');
      if (data.hasOwnProperty('rf_campaign_type'))
        obj.rf_campaign_type = ApiClient.convertToType(data['rf_campaign_type'], 'String');
      if (data.hasOwnProperty('roas_bid'))
        obj.roas_bid = ApiClient.convertToType(data['roas_bid'], 'Number');
      if (data.hasOwnProperty('rta_bid_enabled'))
        obj.rta_bid_enabled = ApiClient.convertToType(data['rta_bid_enabled'], 'Boolean');
      if (data.hasOwnProperty('rta_id'))
        obj.rta_id = ApiClient.convertToType(data['rta_id'], 'String');
      if (data.hasOwnProperty('rta_product_selection_enabled'))
        obj.rta_product_selection_enabled = ApiClient.convertToType(data['rta_product_selection_enabled'], 'Boolean');
      if (data.hasOwnProperty('sales_destination'))
        obj.sales_destination = ApiClient.convertToType(data['sales_destination'], 'String');
      if (data.hasOwnProperty('special_industries'))
        obj.special_industries = ApiClient.convertToType(data['special_industries'], ['String']);
      if (data.hasOwnProperty('virtual_objective_type'))
        obj.virtual_objective_type = ApiClient.convertToType(data['virtual_objective_type'], 'String');
    }
    return obj;
  }
}

/**
 * @member {String} advertiser_id
 */
CampaignCreateBody.prototype.advertiser_id = undefined;

/**
 * @member {String} app_id
 */
CampaignCreateBody.prototype.app_id = undefined;

/**
 * @member {String} app_promotion_type
 */
CampaignCreateBody.prototype.app_promotion_type = undefined;

/**
 * @member {String} bid_type
 */
CampaignCreateBody.prototype.bid_type = undefined;

/**
 * @member {Number} budget
 */
CampaignCreateBody.prototype.budget = undefined;

/**
 * @member {String} budget_mode
 */
CampaignCreateBody.prototype.budget_mode = undefined;

/**
 * @member {Boolean} budget_optimize_on
 * @default false
 */
CampaignCreateBody.prototype.budget_optimize_on = false;

/**
 * @member {String} campaign_app_profile_page_state
 */
CampaignCreateBody.prototype.campaign_app_profile_page_state = undefined;

/**
 * @member {String} campaign_name
 */
CampaignCreateBody.prototype.campaign_name = undefined;

/**
 * @member {String} campaign_product_source
 */
CampaignCreateBody.prototype.campaign_product_source = undefined;

/**
 * @member {String} campaign_type
 */
CampaignCreateBody.prototype.campaign_type = undefined;

/**
 * @member {Boolean} catalog_enabled
 */
CampaignCreateBody.prototype.catalog_enabled = undefined;

/**
 * @member {String} deep_bid_type
 */
CampaignCreateBody.prototype.deep_bid_type = undefined;

/**
 * @member {Boolean} disable_skan_campaign
 */
CampaignCreateBody.prototype.disable_skan_campaign = undefined;

/**
 * @member {String} internal_channel
 */
CampaignCreateBody.prototype.internal_channel = undefined;

/**
 * @member {Boolean} is_advanced_dedicated_campaign
 */
CampaignCreateBody.prototype.is_advanced_dedicated_campaign = undefined;

/**
 * @member {Boolean} is_search_campaign
 */
CampaignCreateBody.prototype.is_search_campaign = undefined;

/**
 * @member {String} objective_type
 */
CampaignCreateBody.prototype.objective_type = undefined;

/**
 * @member {String} operation_status
 * @default 'ENABLE'
 */
CampaignCreateBody.prototype.operation_status = 'ENABLE';

/**
 * @member {String} optimization_goal
 */
CampaignCreateBody.prototype.optimization_goal = undefined;

/**
 * @member {String} plugin_partner
 */
CampaignCreateBody.prototype.plugin_partner = undefined;

/**
 * @member {String} po_number
 */
CampaignCreateBody.prototype.po_number = undefined;

/**
 * @member {String} postback_window_mode
 */
CampaignCreateBody.prototype.postback_window_mode = undefined;

/**
 * @member {String} request_id
 */
CampaignCreateBody.prototype.request_id = undefined;

/**
 * @member {String} rf_campaign_type
 */
CampaignCreateBody.prototype.rf_campaign_type = undefined;

/**
 * @member {Number} roas_bid
 */
CampaignCreateBody.prototype.roas_bid = undefined;

/**
 * @member {Boolean} rta_bid_enabled
 * @default false
 */
CampaignCreateBody.prototype.rta_bid_enabled = false;

/**
 * @member {String} rta_id
 */
CampaignCreateBody.prototype.rta_id = undefined;

/**
 * @member {Boolean} rta_product_selection_enabled
 */
CampaignCreateBody.prototype.rta_product_selection_enabled = undefined;

/**
 * @member {String} sales_destination
 */
CampaignCreateBody.prototype.sales_destination = undefined;

/**
 * @member {Array.<String>} special_industries
 */
CampaignCreateBody.prototype.special_industries = undefined;

/**
 * @member {String} virtual_objective_type
 */
CampaignCreateBody.prototype.virtual_objective_type = undefined;

